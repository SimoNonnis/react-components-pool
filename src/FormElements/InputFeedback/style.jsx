import styled from 'styled-components';

export const ErrorStyled = styled.div`
  color: ${({ theme }) => theme.color.red};
  font-size: ${({ theme }) => theme.base.fontSize};
  margin-top: 7px;
  min-height: 17px;
`;
