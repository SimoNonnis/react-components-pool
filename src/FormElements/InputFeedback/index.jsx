import React from 'react';
import PropTypes from 'prop-types';
import { ErrorStyled } from './style';

const propTypes = {
  error: PropTypes.string,
  className: PropTypes.string,
};
const defaultProps = {
  error: null,
  className: null,
};

const InputFeedback = ({ error, className }) => (
  <ErrorStyled className={className}>{error}</ErrorStyled>
);

InputFeedback.propTypes = propTypes;
InputFeedback.defaultProps = defaultProps;

export default InputFeedback;
