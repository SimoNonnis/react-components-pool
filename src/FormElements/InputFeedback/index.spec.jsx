/* eslint-env jest */
import React from 'react';

import 'jest-styled-components';

import InputFeedback from './';

describe('Test <InputFeedback />', () => {
  it('should render InputFeedback correctly with default props', () => {
    const wrapper = shallow(<InputFeedback />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render InputFeedback correctly with custom props', () => {
    const wrapper = shallow(<InputFeedback error="Error message" className="className" />);
    expect(wrapper).toMatchSnapshot();
  });
});
