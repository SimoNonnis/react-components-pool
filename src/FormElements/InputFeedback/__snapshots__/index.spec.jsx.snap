// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <InputFeedback /> should render InputFeedback correctly with custom props 1`] = `
<styled.div
  className="className"
>
  Error message
</styled.div>
`;

exports[`Test <InputFeedback /> should render InputFeedback correctly with default props 1`] = `
<styled.div
  className={null}
/>
`;
